qiskit/opflow/gradients/hessian.py

```diff
@@ -56,19 +57,15 @@ def convert(self,
 
         Returns:
             OperatorBase: An operator whose evaluation yields the Hessian
-
-        Raises:
-            ValueError: If `params` is not set.
         """
         # if input is a tuple instead of a list, wrap it into a list
-        if params is None:
-            raise ValueError("No parameters were provided to differentiate")
-
         if isinstance(params, (ParameterVector, list)):
             # Case: a list of parameters were given, compute the Hessian for all param pairs
             if all(isinstance(param, ParameterExpression) for param in params):
-                return ListOp(
-                    [ListOp([self.convert(operator, (p0, p1)) for p1 in params]) for p0 in params])
+                return ListOp([self.convert(operator, (param_i, param_j))
```

Comment:
This changes the output type if one passes a `ParameterVector` as `params`. Before, it was a matrix and with this change it is a matrix. I think it should remain a matrix since that's what you would expect when asking for the Hessian with respect to a vector-like parameter. 

I agree though that there's some redundancy in computing all entries of the symmetric matrix, but if we want to solve that we might want to consider introducing our proper matrix type or subtype of a ListOp and not do it in this PR.

Reply:
thanks @Cryoris, I got your point.
Hessian returns matrix again, and unit test modified accordingly.